src/cmd/ksh93/bltins/misc.c

```diff
@@ -218,6 +218,11 @@ int    b_dot_cmd(int n,char *argv[],Shbltin_t *context)
 	struct checkpt buff;
 	Sfio_t *iop=0;
 	short level;
+	Optdisc_t disc;
```

Comment:
It looks like the `optinit()` macro is supposed to be used to do this initialization. But this is another example where we should replace the AST code with standard functions. In this case `getopt_long()`. So I'm going to let it stand.